classes/data-stores/ActionScheduler_DBStore.php

```diff
@@ -596,8 +603,11 @@ protected function get_date_gmt( $action_id ) {
 	 */
 	public function stake_claim( $max_actions = 10, \DateTime $before_date = null, $hooks = array(), $group = '' ) {
 		$claim_id = $this->generate_claim_id();
+
+		$this->before_dates_by_claim_id[ $claim_id ] = $before_date;
 		$this->claim_actions( $claim_id, $max_actions, $before_date, $hooks, $group );
 		$action_ids = $this->find_actions_by_claim_id( $claim_id );
+		unset( $this->before_dates_by_claim_id[ $claim_id ] );
```

Comment:
@barryhughes This looks good to me. A small nitpick is that `before_dates_by_claim_id` will never have more than one element in the array.

Reply:
You're right, I over-complicated things there. Will simplify down to a scalar :+1: